src/mlpack/tests/string_utility_test.cpp

```diff
@@ -0,0 +1,176 @@
+  boost::string_view strview = "hello how are you" ;
+  SplitByChar obj(" ");
+  boost::string_view token;
+  token = obj(strview);
+  while (!token.empty())
+  {
+    tokens.push_back(token);
+    token = obj(strview);
+  }
+  BOOST_REQUIRE_EQUAL(tokens[0], "hello");
+  BOOST_REQUIRE_EQUAL(tokens[1], "how");
+  BOOST_REQUIRE_EQUAL(tokens[2], "are");
+  BOOST_REQUIRE_EQUAL(tokens[3], "you");
+}
+
+/**
+* Test Dictionary encoding for characters using lamda function.
+*/
+BOOST_AUTO_TEST_CASE(DictionaryEncodingCharTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "GACCA";
+  arr[1] = "ABCABCD";
+  arr[2] = "GAB";
+  arma::sp_mat output;
+  data::DictionaryEncoding en;
+  // Passing a empty string to encode characters
+  en.Encode(arr, output, [](boost::string_view& str) {
+      if (str.empty())
+        return str;
+      boost::string_view retval = str.substr(0, 1);
+      str.remove_prefix(1);
+      return retval;
+  });
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>>& maps = en.Mappings();
+  // Checking that everything is mapped to different numbers.
+  std::unordered_map<size_t, size_t>cnt;
+  for (auto it = maps.begin(); it != maps.end(); it++)
+  {
+    cnt[it->second]++;
+    // Every token should be mappend only once.
+    BOOST_REQUIRE_EQUAL(cnt[it->second], 1);
+  }
+}
+
+/**
+* Testing the functionality of copy constructor.
+*/
+BOOST_AUTO_TEST_CASE(CopyConstructorCharTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "hello how are you";
+  arr[1] = "i am good";
+  arr[2] = "Good how are you";
+  arma::sp_mat output;
+  data::DictionaryEncoding en;
+  en.Encode(arr, output, SplitByChar(" "));
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>>& maps = en.Mappings();
+  data::DictionaryEncoding en2 = en;
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>>& maps2 = en2.Mappings();
+  // Comparing both of them.
+  BOOST_REQUIRE_EQUAL(maps2 == maps, true);
+}
+
+/**
+* Test Dictionary encoding for output with no padding.
+*/
+BOOST_AUTO_TEST_CASE(DictionaryEncodingNoPaddingTest)
+{
+  std::vector<string>arr(2);
+  arr[0] = "GACCA";
+  arr[1] = "GAB";
+  data::DictionaryEncoding en;
+  std::vector<std::vector<size_t> > output;
+  en.Encode(arr, output, [](boost::string_view& str) {
+      if (str.empty())
+        return str;
+      boost::string_view retval = str.substr(0, 1);
+      str.remove_prefix(1);
+      return retval;
+  });
+  std::vector<std::vector<size_t>> req_output;
+  std::vector<size_t> temp(5);
+  temp[0] = 1;
+  temp[2] = temp[3] = 3;
+  temp[1] = temp[4] = 2;
+  req_output.push_back(temp);
+  temp.clear();
+  temp.push_back(1);
+  temp.push_back(2);
+  temp.push_back(4);
+  req_output.push_back(temp);
+  for (size_t i = 0; i < 2; i++)
+  {
+    for (size_t j = 0; j < req_output[i].size(); j++)
+    {
+      BOOST_REQUIRE_EQUAL(req_output[i][j], output[i][j]);
```

Comment:
I think `BOOST_REQUIRE_EQUAL(req_output, output);` should work.

Reply:
BOOST_REQUIRE_EQUAL doesn't support this and hence it  throws error and floods by screen.

Reply:
Probably, it doesn't know how to write the class to a stream. Okay, let it be as is. Could you get rid of the constant (`2`) in the loop?
```
for (size_t i = 0; i < 2; i++)
```